### ISSUE TITLE
Setting the "header" for a list

### ISSUE THREAD

**Issue by pecigonzalo** [dhall-lang/dhall-haskell#2028]
Im using Dhall to generate Kuberentes and I believe it would be great to have some way to denote a "header" for list.

For example 
```dhall
let this = [ { foo = "this", bar = "that" }, { foo = "fizz", bar = "buzz" } ]

in  this
```
Renders
```
- bar: that
  foo: this
- bar: buzz
  foo: fizz
```
I would be great to have some semantics to say "I want foo as my `-` value". This would be really helpful in container definitions, as it would mean we can pin either `image` or `name` or something meaningful to the list `-`. This gets generally set to `args` or some other setting which does not identify the container.

Example
```
let this : Type = { foo |= Text, bar = Text }
```

This seems like a minor annoyance, but when you have a list with multiple containers and each container with a long list of defined settings, it gets really hard to parse them.

An alternative idea would be to follow the order set by the Type definition if one is specified, which would not work for the previous example, but would allow me to define a type to explicitly set an order.

PS: I think it would be great for Kubernetes to use `servicename:` and define these values inside of the key, as its done in Docker Compose.

**Comment by Gabriel439** [comment]
This appears to be the same issue as https://github.com/dhall-lang/dhall-haskell/issues/2028

**Comment by pecigonzalo** [comment]
Thanks! I was not sure if this was something to be implemented in the core code, as the 2nd option would apply to all output.

The link seems to self reference this issue.

**Comment by Gabriel439** [comment]
Whoops!  I fixed the link

**Comment by pecigonzalo** [comment]
Reading the issue, that maps perfectly to that second suggestion, and would be a more complete solution. I'm happy to close this one and continue over there, but I defer to your judgment.

**Comment by Gabriel439** [comment]
Alright, then I'll close this out in favor of the other issue
